tests/Collection/BuilderCollectionFunctionalTest.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+namespace MongoDB\Tests\Collection;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+
+class BuilderCollectionFunctionalTest extends FunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->collection->insertMany([['x' => 1], ['x' => 2], ['x' => 2]]);
+    }
+
+    public function testAggregate(): void
+    {
+        $this->markTestSkipped('Not supported yet');
+    }
+
+    public function testBulkWriteDeleteMany(): void
+    {
+        $result = $this->collection->bulkWrite([
+            [
+                'deleteMany' => [
+                    Query::query(x: Query::gt(1)),
```

Comment:
Not necessary for this PR, but I just came to think that the following might be a little easier on the eyes:
```suggestion
                    new Query(x: Query::gt(1)),
```

Alternatively, if we have a generic builder class:
```suggestion
                    new Builder::query(x: Query::gt(1)),
```

For some reason, the `Query::query(...)` part irks me a bit 🤔

Reply:
Sure, if we can create functions for all the factories that would shorten the syntax.